src/tests/multihost/sssd/testlib/common/expectsudo.py

```diff
@@ -0,0 +1,406 @@
+import os
+import sys
+import time
+
+import pexpect
+from pexpect import pxssh, ExceptionPexpect, TIMEOUT, EOF, spawn
+from pexpect.pxssh import ExceptionPxssh
+from sssd.testlib.common.exceptions import OSException
+
+
+class pexpect_ssh(spawn):
```

Comment:
pexpect_sudo? This class looks like it deals more with ssh then sudo, I am confused. There is already `pexpect_ssh` in `expect.py` which deals with ssh, isn't that sufficient?

Reply:
You are right, this utility involves ssh + regexes to carry out the tasks, the existing `expect.py` is bugged, cannot work with the tests that are required by the sudo-sssd test cases. Therefore I built a pexpect based utility that is highly optimized and covers all edge cases.  `expect.py` is used in so many places, it was quite risky to replace the code in `expect.py`, thats why there's a separate file `expectsudo.py`.

Reply:
IMHO if expect.py is  bugged, it needs to be fixed. If there are tests that relies on buggy behavior, these needs to be fixed as well.

Reply:
@pbrezina I will track the fixing of expect.py as a separate task internally.